designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NodesApiEndpoints.scala

```diff
@@ -497,35 +742,112 @@ object NodesApiEndpoints {
+      BadRequest,
+      plainBody[MalformedTypingResult]
+        .example(
+          Example.of(
+            summary = Some("Malformed TypingResult sent in request"),
+            value = MalformedTypingResult(
+              "Couldn't decode value 'WrongType'. Allowed values: 'TypedUnion,TypedDict,TypedObjectTypingResult,TypedTaggedValue,TypedClass,TypedObjectWithValue,TypedNull,Unknown"
+            )
+          )
+        )
+    )
+
+  private val noProcessingTypeExample: EndpointOutput.OneOfVariant[NoProcessingType] =
+    oneOfVariantFromMatchType(
+      NotFound,
+      plainBody[NoProcessingType]
+        .example(
+          Example.of(
+            summary = Some("ProcessingType type: {processingType} not found"),
+            value = NoProcessingType("'processingType'")
+          )
+        )
+    )
+
 }
 
 object TypingDtoSchemas {
 
+  import pl.touk.nussknacker.engine.api.typed.TypingType
   import pl.touk.nussknacker.engine.api.typed.TypingType.TypingType
   import pl.touk.nussknacker.engine.api.typed.typing._
-  import pl.touk.nussknacker.engine.api.typed.TypingType
   import sttp.tapir.Schema.SName
   import sttp.tapir.SchemaType.SProductField
   import sttp.tapir.{FieldName, Schema, SchemaType}
 
-  implicit lazy val typingResult: Schema[TypingResult] = Schema.derived
-
-  implicit lazy val singleTypingResultSchema: Schema[SingleTypingResult] =
-    Schema.derived.hidden(true)
+  implicit lazy val typingResult: Schema[TypingResult] = {
+    Schema(
+      SchemaType.SCoproduct(
+        List(
+          unknownSchema.hidden(true),
+          unknownSchema,
+          typedNullSchema,
+          typedObjectTypingResultSchema,
+          typedDictSchema,
+          typedObjectSchema,
+          typedClassSchema,
+          typedUnionSchema,
+          typedTaggedSchema
+        ),
+        None
+      ) {
+        case Unknown                              => Some(SchemaWithValue(unknownSchema, Unknown))
+        case TypedNull                            => Some(SchemaWithValue(typedNullSchema, TypedNull))
+        case typedObject: TypedObjectTypingResult => Some(SchemaWithValue(typedObjectTypingResultSchema, typedObject))
+        case typedDict: TypedDict                 => Some(SchemaWithValue(typedDictSchema, typedDict))
+        case typedWithValue: TypedObjectWithValue => Some(SchemaWithValue(typedObjectSchema, typedWithValue))
+        case typedClass: TypedClass               => Some(SchemaWithValue(typedClassSchema, typedClass))
+        case union: TypedUnion                    => Some(SchemaWithValue(typedUnionSchema, union))
+        case tagged: TypedTaggedValue             => Some(SchemaWithValue(typedTaggedSchema, tagged))
+      }
+    )
+  }
 
+  implicit lazy val singleTypingResultSchema: Schema[SingleTypingResult]   = Schema.derived
   implicit lazy val additionalDataValueSchema: Schema[AdditionalDataValue] = Schema.derived
 
+  object TypedObjectTypingResultSchemaHelper {
+    sealed trait types
```

Comment:
It'd be good to explain why we do it like that. So, let's say that in json schema v4 (which is used by the current version of Tapir) there is no way to declare discriminator, so we have to workaround this problem by adding it manually.